datashader/tests/test_datatypes.py

```diff
@@ -886,3 +897,11 @@ def test_transpose(self):
     @pytest.mark.skip(reason="transpose with numpy array elements seems not supported")
     def test_transpose_frame(self):
         pass
+
+    @pytest.mark.skipif(Version(pd.__version__) >= Version("2.2"), reason="Stopped working in Pandas 2.2")
+    def test_merge_on_extension_array(self, data):
```

Comment:
I have no idea why these two tests started failing. It does not seem to be because of a change to the function itself.

Reply:
I will open an issue after this PR is merged so we don't forget it.

Reply:
It seems to be related to a change in `pd.merge`, A MRE is this:

``` python
import pandas as pd
from datashader.datatypes import RaggedArray

key = RaggedArray([[0, 1], [1, 2, 3, 4]], dtype='float64')
df = pd.DataFrame({"key": key, "val": [1, 2]})

pd.merge(df, df, on='key')
```

Pandas 2.1:
![image](https://github.com/holoviz/datashader/assets/19758978/8ffe9a84-2656-4462-abbd-31484f06d5a3)



Pandas 2.2:
`ValueError: operands could not be broadcast together with shapes (4,) (2,) `

Reply:
Maybe because of https://github.com/pandas-dev/pandas/pull/56523

Reply:
It is because of that PR:

``` python
import pandas as pd
from datashader.datatypes import RaggedArray

key = RaggedArray([[0, 1], [1, 2, 3, 4]], dtype='float64')
pd.Index(key).is_monotonic_increasing()
```

Reply:
Opened an issue upstream: https://github.com/pandas-dev/pandas/issues/57316